src/app/containers/FrontPageStoryRows/index.jsx

```diff
@@ -1,49 +1,141 @@
 import React from 'react';
 import Grid from '@bbc/psammead-grid';
-import { node, arrayOf, shape, any } from 'prop-types';
+import { arrayOf, shape, bool, oneOf } from 'prop-types';
+import styled, { css } from 'styled-components';
+
+import {
+  GEL_GROUP_1_SCREEN_WIDTH_MIN,
+  GEL_GROUP_2_SCREEN_WIDTH_MAX,
+} from '@bbc/gel-foundations/breakpoints';
+import {
+  GEL_SPACING_DBL,
+  GEL_SPACING_TRPL,
+} from '@bbc/gel-foundations/spacings';
+import { StoryPromoLi } from '@bbc/psammead-story-promo-list';
 import {
   topStoryColumns,
   leadingStoryColumns,
   regularStoryColumns,
 } from './storyColumns';
+import { storyItem } from '#models/propTypes/storyItem';
+import BulletinContainer from '../Bulletin';
+import StoryPromoContainer from '../StoryPromo';
+
+const StoryPromoListItem = styled(StoryPromoLi)`
+  ${({ isBulletin, isFirstPromo }) =>
+    isBulletin &&
+    css`
+      @media (min-width: ${GEL_GROUP_1_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
+        ${isFirstPromo
+          ? `padding-bottom: ${GEL_SPACING_TRPL};`
+          : `padding: ${GEL_SPACING_DBL} 0 ${GEL_SPACING_TRPL};`}
+      }
+    `}
+`;
+
+const isBulletin = item =>
+  item.contentType === 'TVBulletin' || item.contentType === 'RadioBulletin';
+
+const renderPromo = (
+  item,
+  promoType,
+  optional = { isFirstSection: false, displayImage: true },
+) => {
+  const lazyLoadImage = promoType === 'top' && optional.isFirstSection; // don't lazy load image if it is a top story
+  const renderedPromo = isBulletin(item) ? (
+    <BulletinContainer item={item} lazyLoadImage={lazyLoadImage} />
+  ) : (
+    <StoryPromoContainer
+      item={item}
+      promoType={promoType}
+      lazyLoadImage={lazyLoadImage}
+      displayImage={optional.displayImage}
+    />
+  );
+
+  return renderedPromo;
+};
 
-export const TopRow = ({ story }) => (
-  <Grid item columns={topStoryColumns}>
-    {story}
+export const TopRow = ({ stories, isFirstSection, isSingleStory, dir }) => (
```

Comment:
It's not clear that the `isSingleStory` is referring to a single story in the whole section and not just a single story in the row.  It'd be good to rename it to something more descriptive. Perhaps `sectionHasSingleStory`?